internal/store.go

```diff
@@ -25,7 +29,20 @@ func (s *Store) GetSDK() pkg.CloudSDK {
 	return s.sdk
 }
 
+var (
+	claimOrganizationID = "organization_id"
+)
+
 // GetOrganizationID returns the current organization ID
-func (s *Store) GetOrganizationID() string {
-	return strings.Split(s.clientID, "organization_")[1]
+func (s *Store) GetOrganizationID(ctx context.Context) string {
+	s.Lock()
+	defer s.Unlock()
+	if s.organizationID == "" {
+		introspection, err := s.tp.IntrospectToken(ctx)
+		if err != nil {
+			panic(err)
```

Comment:
Are you sure you want to panic so deep in the code? I understand it's tempting to have a easy Get value interface, but without a compelling reason I'd say it's better to do proper error handling so the entire application doesn't crash mid operation.

Reply:
For me one thing to consider is "how likely is it to fail?"

Because when you panic you're not just impacting the user that has the error, but all users that are concurrently using the application. So one user that has a problem and keeps pressing refresh, can cause instability for all the other users of the application if there is a panic like this.

Adding if statements to catch the error is a pain in the ass, but it is usually worth it if the underlying function makes http calls, for example.